### PR TITLE
Bk/refactors and tweaks for 2.0.0

### PR DESCRIPTION
**Pull request by bryankeller**
## Details

Continuing to work toward a 2.0.0 release, I've made several API improvements and tweaks. Things are broken up by commit, if it helps.

From the CHANGELOG.md:

- Renamed `viewModel` to `content`, since it's a less-overloaded term
- Changed `monthDayInsets` to take `NSDirectionalEdgeInsets`, rather than `UIEdgeInsets`
- Un-nested some types to make them fit better with an upcoming SwiftUI API
- Updated the default `HorizontalMonthLayoutOptions` to use a `restingPosition` of `.atLeadingEdgeOfEachMonth`, which is probably what most people want
- Added `MonthsLayout.vertical` and `MonthsLayout.horizontal` as more concise alternatives to `MonthsLayout.vertical(options: .init())` and `MonthsLayout.horizontal(options: .init())`, respectively
- Make more publicly-exposed types conform to `Hashable`, because why not - maybe API consumers want to stick things in a `Set` or `Dictionary`.
- Allow `CalendarItemViewRepresentable` to have no `Content` type if the conforming view does not depend on any variable data

## Related Issue

N/A

## Motivation and Context

API tweaks and improvements for the 2.0.0 release

## How Has This Been Tested

Example app

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: Sources/Internal/Hasher+CGRect.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import Foundation
+
+extension Hasher {
+
+  mutating func combine(_ rect: CGRect) {
+    combine(rect.origin.x)
+    combine(rect.origin.y)
+    combine(rect.size.width)
+    combine(rect.size.height)
+  }
```
Comment (bryankeller) [inline Sources/Internal/Hasher+CGRect.swift:25]: huge help tbh

File: Sources/Public/MonthLayoutContext.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// The layout context for all of the views contained in a month, including frames for days, the month header, and days-of-the-week
+/// headers. Also included is the bounding rect (union) of those frames. This can be used in a custom month background view to draw
+/// the background around the month's foreground views.
+public struct MonthLayoutContext: Hashable {
```
Comment (bryankeller) [inline Sources/Public/MonthLayoutContext.swift:19]: just moved out of CalendarViewContent

File: Sources/Public/DaysOfTheWeekRowSeparatorOptions.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+
+/// Used to configure the days-of-the-week row's separator.
+public struct DaysOfTheWeekRowSeparatorOptions: Hashable {
```
Comment (bryankeller) [inline Sources/Public/DaysOfTheWeekRowSeparatorOptions.swift:19]: just moved out of CalendarViewContent

File: Sources/Public/DayRangeLayoutContext.swift [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import CoreGraphics
+
+/// The layout context for a day range, containing information about the frames of days in the day range and the bounding rect (union)
+/// of those frames. This can be used in a custom day range view to draw the day range in the correct location.
+public struct DayRangeLayoutContext: Hashable {
```
Comment (bryankeller) [inline Sources/Public/DayRangeLayoutContext.swift:20]: just moved out of CalendarViewContent

File: Sources/Public/OverlaidItemLocation.swift [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Represents the location of an item that can be overlaid.
+public enum OverlaidItemLocation: Hashable {
```
Comment (bryankeller) [inline Sources/Public/OverlaidItemLocation.swift:19]: just moved out of CalendarViewContent

File: Sources/Public/OverlayLayoutContext.swift [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import CoreGraphics
+
+/// The layout context for an overlaid item, containing information about the location and frame of the item being overlaid, as well as the
+/// bounds available to the overlay item for drawing and layout.
+public struct OverlayLayoutContext: Hashable {
```
Comment (bryankeller) [inline Sources/Public/OverlayLayoutContext.swift:20]: just moved out of CalendarViewContent

**Review by brynbodayle — APPROVED** [review]

**Pull request merged**